### DISCUSSION TITLE
[HTTP]:500 - [CorrelationId] - Help with diagnosis

### DISCUSSION THREAD

**Discussion by Parmstig** [microsoft-search/pnp-modern-search#3204] [Q&A]
We used Modern Search as a main function to gather content on a Intranet with around 4000 users. Its used everywhere as both People Galleries, and showing different types of content from pages. HandlebarJS have been used for styling.

We get alot of HTTP:500 errors, is often per user and not in the same place for everyone, the queries are pretty long and complex but have been optimized. But the actual query does not really matter since the error can occur at random, but mostly is something like this:
{searchTerms} Path:"https://[redacted].sharepoint.com/sites/operating-info" ContentTypeId:0x0101009D1CB255DA76424F860D91F20E6C4118000AE549343BF9C84A8DAF37A6619BD849*

The error is always a response with "unknown error"
{"odata.error":{"code":"-1, Microsoft.Office.Server.Search.REST.SearchServiceException","message":{"lang":"sv-SE","value":"Ett ok\u00e4nt fel uppstod."}}}

* Handlebar/HTML is very custom
* Its used against multilang and many audiences

What can i try or do to understand what the problem is?

**Comment by kasperbolarsen** [comment]
Hi Tony,
First step should be to check if this is a PnP Modern Search issue at all. Based on the description I will expect it to be a general search issue/API issue.
Would it be possible for you to run the same query in code or set up a classic search page and check if the same error occurs?

**Reply by Parmstig** [reply]
I have tried it with both classic search and sending query thru powershell. And the hard part is to reproduce the error, its only happening in MSWP.

**Reply by kasperbolarsen** [reply]
And only for your heavily customized People Layout? Not in the default People Layout?

**Comment by kasperbolarsen** [comment]
You are on version 4.9.* , right ?

**Reply by Parmstig** [reply]
Yes i updated yesterday. Im trying a script now and it works every time - i will try to run in paralell several times.
Connect-PnPOnline -Url "xxx365.sharepoint.com" -DeviceLogin

$query = "Path:https://xx365.sharepoint.com/sites/hrportal ContentTypeId:0x0101009D1CB255DA76424F860D91F20E6C411800C86B2CA1BD562F4C9CA39252BF9FD7D1* (SPTranslationLanguage:{CurrentDisplayLanguage} OR (NOT SPTranslatedLanguages:{CurrentDisplayLanguage} AND NOT SPIsTranslation:true)) PromotedState:2"
Submit-PnPSearchQuery -Query $query
